### PR TITLE
Fixed repo link in generated comment for cross repository dependency

### PR DESCRIPTION
**Pull request by bhalbright**
Fix issue #9765. When adding or removing a dependency, a comment is generated on each issue. The comment when displayed in the UI assumes that the issue is in the same repository, but with cross repository dependencies, they are not. Fixed to show the full name of the issue and links to the correct issue.

### CODE REVIEW

**Comment by bhalbright** [comment]
Before:
image

After:
image

File: templates/repo/issue/view_content/comments.tmpl [hunk 1]
```diff
@@ -279,7 +279,7 @@
 			</span>
 			<div class="detail">
 				<span class="octicon octicon-plus"></span>
-				<span class="text grey"><a href="{{$.RepoLink}}/issues/{{.DependentIssue.Index}}">#{{.DependentIssue.Index}} {{.DependentIssue.Title}}</a></span>
+				<span class="text grey"><a href="{{AppSubUrl}}/{{.DependentIssue.Repo.OwnerName}}/{{.DependentIssue.Repo.Name}}/issues/{{.DependentIssue.Index}}">{{.DependentIssue.Repo.FullName}}#{{.DependentIssue.Index}} - {{.DependentIssue.Title}}</a></span>
```
Comment (guillep2k) [inline templates/repo/issue/view_content/comments.tmpl]: I think that `.DependentIssue` could be `nil` if the user has no permission to read the other repo? Perhaps an `{{if .DependentIssue}}` should be used.

Comment (lunny) [reply templates/repo/issue/view_content/comments.tmpl]: Just use `{{.DependentIssue.HTMLURL}}`.

File: templates/repo/issue/view_content/comments.tmpl [hunk 2]
```diff
@@ -293,7 +293,7 @@
 			</span>
 			<div class="detail">
 				<span class="text grey octicon octicon-trashcan"></span>
-				<span class="text grey"><a href="{{$.RepoLink}}/issues/{{.DependentIssue.Index}}">#{{.DependentIssue.Index}} {{.DependentIssue.Title}}</a></span>
+				<span class="text grey"><a href="{{AppSubUrl}}/{{.DependentIssue.Repo.OwnerName}}/{{.DependentIssue.Repo.Name}}/issues/{{.DependentIssue.Index}}">{{.DependentIssue.Repo.FullName}}#{{.DependentIssue.Index}} - {{.DependentIssue.Title}}</a></span>
```
Comment (lunny) [inline templates/repo/issue/view_content/comments.tmpl]: As above.

File: templates/repo/issue/view_content/comments.tmpl (hunk 1 above)
Comment (bhalbright) [reply templates/repo/issue/view_content/comments.tmpl]: updated both, thanks

File: templates/repo/issue/view_content/comments.tmpl (hunk 2 above)
Comment (bhalbright) [reply templates/repo/issue/view_content/comments.tmpl]: updated, thanks

**Comment by lunny** [comment]
And I think maybe you should add `if isCrossRef` so that if it's not a cross repository, we should still keep the old text.

**Review by guillep2k — APPROVED** [review]

File: templates/repo/issue/view_content/comments.tmpl [hunk 3]
```diff
@@ -277,10 +277,12 @@
...
 				{{$.i18n.Tr "repo.issues.dependency.added_dependency" .Poster.HomeLink (.Poster.GetDisplayName|Escape) $createdStr | Safe}}
 			</span>
-			<div class="detail">
-				<span class="octicon octicon-plus"></span>
-				<span class="text grey"><a href="{{$.RepoLink}}/issues/{{.DependentIssue.Index}}">#{{.DependentIssue.Index}} {{.DependentIssue.Title}}</a></span>
-			</div>
+			{{if .DependentIssue}}
+				<div class="detail">
+					<span class="octicon octicon-plus"></span>
+					<span class="text grey"><a href="{{.DependentIssue.HTMLURL}}">{{.DependentIssue.Repo.FullName}}#{{.DependentIssue.Index}} - {{.DependentIssue.Title}}</a></span>
```
Comment (6543) [inline templates/repo/issue/view_content/comments.tmpl]: If current repo != dependentissue.repo ...

Comment (6543) [reply templates/repo/issue/view_content/comments.tmpl]: (Only show repo name if needed)

**Review by 6543 — CHANGES_REQUESTED** [review]
as last commit

File: templates/repo/issue/view_content/comments.tmpl (hunk 3 above)
Comment (bhalbright) [reply templates/repo/issue/view_content/comments.tmpl]: thanks @6543 and @lunny I changed as suggested:

image

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Comment by zeripath** [comment]
Please send backport

**Comment by 6543** [comment]
created bakport for you @bhalbright

**Comment by bhalbright** [comment]
> created bakport for you @bhalbright (#9935)

Thanks @6543 I wasn't going to have time this week, appreciate it.

**Pull request merged**